specs/windows/signature.table

```diff
@@ -0,0 +1,17 @@
+table_name("signature")
+description("File (executable, bundle, installer, disk) code signing status.")
+schema([
+    Column("path", TEXT, "Must provide a path or directory", required=True),
+    Column("original_program_name", TEXT, "The original program name that the publisher has signed"),
+    Column("serial_number", TEXT, "The certificate serial number"),
+    Column("issuer_name", TEXT, "The certificate issuer name"),
+    Column("subject_name", TEXT, "The certificate subject name"),
+    Column("result", TEXT, "The signature check result")
+])
```

Comment:
Can we morph fields from darwin or windows such that we can have a unified `signature` table?
```
schema([
    Column("path", TEXT, "Must provide a path or directory", required=True),
    Column("signed", INTEGER, "1 If the file is signed else 0"),
    Column("identifier", TEXT, "The signing identifier sealed into the signature"),
    Column("cdhash", TEXT, "SHA1 hash of the application Code Directory"),
    Column("team_identifier", TEXT, "The team signing identifier sealed into the signature"),
    Column("authority", TEXT, "Certificate Common Name"),
])
```

- `path` cool!
- `original_program_name` can this be called `identifier` on both?
- `serial_number`, `issuer_name`, `subject_name` can be extended schema on Windows for now.
- `cdhash` does this checksum have an authenticode equivalent?
- `result` can stay on both, maybe on darwin we fill this in with "signed" or "unknown"
- `signed` being a boolean should be included here, it should indicate if the binary is properly signed.

Reply:
The reason that the build is breaking in the CI is because there are two specfiles named `signature`. This is not allowed.

Reply:
**original_program_name**: I have renamed this to "identifier".
**cdhash**: There is something but to grab it you need to parse the PE header and locate the PKCS#7 structure that contains the hash. There is really no easy way to calculate it without a custom tool (and it's hard to use this value in a meaningful way) so I skipped it.
**result**: I'll add this column to the darwin schema, and set it to "signed" when "signed" is 1 and to "unknown" when it is 0.
**signed**: Should this value be 1 only when the signature is both valid and trusted?

What should I do with the remaining fields? Should both specfiles (darwin and Windows) contain the exact same columns?

Reply:
Leave the fields blank if there are no meaningful compromises/merging options. 

You can add "extended" schema like:
```
extended_schema(WINDOWS, [
    Column("friendly_name", TEXT, "The friendly display name of the interface."),
])
```

The first option is the requirement. These columns will be set to `HIDDEN` on platforms that do not match. Such that a `SELECT *` does not return them.

Your options for evaluators right now are: WINDOWS, LINUX, POSIX, DARWIN, FREEBSD. They are small functions defined in `gentable.py`.